File: Search/c++/11_카드정렬하기.cpp
// fast campus 강의
// https://www.acmicpc.net/problem/1715
// 1

#include <iostream>
#include <vector>
#include <queue>

using namespace std;

int N;
priority_queue<int, vector<int>, greater<int>> pq;

int main() {
	cin >> N;
	for (int i = 0; i < N; i++) {
		int tmp;
		cin >> tmp;
		pq.push(tmp);
	}

	int result = 0;
	while (pq.size() > 1) {
		int a = pq.top();
		pq.pop();
		int b = pq.top();
		pq.pop();
		int sum_value = a + b;
		result += sum_value;
		pq.push(sum_value);
	}
	cout << result;
}

File: Search/python/11_카드정렬하기.py
# fast campus 강의
# https://www.acmicpc.net/problem/1715
# 1

import heapq

n = int(input())

heap = []
for _ in range(n):
    heapq.heappush(heap, int(input()))

result = 0
while len(heap) > 1:
    a = heapq.heappop(heap)
    b = heapq.heappop(heap)
    sum_value = a + b
    result += sum_value
    heapq.heappush(heap, a+b)
print(result)